src/coreComponents/physicsSolvers/solidMechanics/SolidMechanicsEFEMKernelsHelper.hpp

```diff
@@ -0,0 +1,239 @@
+                                    arraySlice1d< real64 const > const & tVec2,
+                                    integer ( & heavisideFun )[NUM_NODES],
+                                    real64 ( & dNdX )[NUM_NODES][3] )
+{
+  //GEOSX_MARK_FUNCTION;
+
+  // Fill in compatibility operator
+
+  // 1. construct mvector sum(dNdX(a) * H(a)) value for each Gauss point
+
+  real64 mVec[3];
+  LvArray::tensorOps::fill< 3 >( mVec, 0 );
+  for( integer a=0; a<NUM_NODES; ++a )
+  {
+    mVec[0] -= dNdX[a][0] * heavisideFun[a];
+    mVec[1] -= dNdX[a][1] * heavisideFun[a];
+    mVec[2] -= dNdX[a][2] * heavisideFun[a];
+  }
+
+
+  // 2. fill in the operator itself
+
+  LvArray::tensorOps::fill< 6, 3 >( compMatrix, 0 );
+
+  real64 nDmSym[3][3], t1DmSym[3][3], t2DmSym[3][3];
+
+  // sym(n dyadic m)
+  LvArray::tensorOps::Rij_eq_AiBj< 3, 3 >( nDmSym, mVec, nVec );
+  LvArray::tensorOps::Rij_add_AiBj< 3, 3 >( nDmSym, nVec, mVec );
+  LvArray::tensorOps::scale< 3, 3 >( nDmSym, 0.5 );
+
+  // sym(n dyadic t1) and sym (n dyadic t2)
+  LvArray::tensorOps::Rij_eq_AiBj< 3, 3 >( t1DmSym, mVec, tVec1 );
+  LvArray::tensorOps::Rij_add_AiBj< 3, 3 >( t1DmSym, tVec1, mVec );
+  LvArray::tensorOps::scale< 3, 3 >( t1DmSym, 0.5 );
+
+  LvArray::tensorOps::Rij_eq_AiBj< 3, 3 >( t2DmSym, mVec, tVec2 );
+  LvArray::tensorOps::Rij_add_AiBj< 3, 3 >( t2DmSym, tVec2, mVec );
+  LvArray::tensorOps::scale< 3, 3 >( t2DmSym, 0.5 );
+
+  int VoigtIndex;
+
+  for( int i=0; i < 3; ++i )
+  {
+    for( int j=0; j < 3; ++j )
+    {
+      if( i == j )
+      {
+        VoigtIndex = 1;
+      }
+      else
+      {
+        VoigtIndex = 6 - i - j;
+      }
+      compMatrix[VoigtIndex][0] += nDmSym [i][j];
+      compMatrix[VoigtIndex][1] += t1DmSym[i][j];
+      compMatrix[VoigtIndex][2] += t2DmSym[i][j];
+    }
+  }
+}
+
+GEOSX_HOST_DEVICE
+GEOSX_FORCE_INLINE
+void assembleEquilibriumOperator( real64 ( & eqMatrix )[3][6],
+                                  arraySlice1d< real64 const > const nVec,
+                                  arraySlice1d< real64 const > const tVec1,
+                                  arraySlice1d< real64 const > const tVec2,
+                                  real64 const hInv )
+{
+  LvArray::tensorOps::fill< 3, 6 >( eqMatrix, 0 );
+
+  // Define the dyadic producs
+  // nDn := nVec_dyadic_nVec,
+  // t1DnSym:= sym(tVec1_dyadic_nVec)
+  // t2DnSym:= sym(tVec2_dyadic_nVec)
+  real64 nDn[3][3], t1DnSym[3][3], t2DnSym[3][3];
+
+  // n dyadic n
+  LvArray::tensorOps::Rij_eq_AiBj< 3, 3 >( nDn, nVec, nVec );
+
+  // sym(n dyadic t1) and sym (n dyadic t2)
+  LvArray::tensorOps::Rij_eq_AiBj< 3, 3 >( t1DnSym, nVec, tVec1 );
+  LvArray::tensorOps::Rij_add_AiBj< 3, 3 >( t1DnSym, tVec1, nVec );
+  LvArray::tensorOps::scale< 3, 3 >( t1DnSym, 0.5 );
+
+  LvArray::tensorOps::Rij_eq_AiBj< 3, 3 >( t2DnSym, nVec, tVec2 );
+  LvArray::tensorOps::Rij_add_AiBj< 3, 3 >( t2DnSym, tVec2, nVec );
+  LvArray::tensorOps::scale< 3, 3 >( t2DnSym, 0.5 );
```

Comment:
You should just write these out.
```suggestion
  real64 nVec2[3];
  for( int i=0; i<3; ++i )
  {
    nVec2[i] = 0.5*nVec[i];
  }
  for( int i=0; i<3; ++i )
  {
    for( int j=0; j<3; ++j )
    {
      t1DnSym[i][j] = nVec2[i] * tVec1[j] + tVec1[i] * nVec2[j];
      t2DnSym[i][j] = nVec2[i] * tVec2[j] + tVec2[i] * nVec2[j];
    }
  }
```
...Or maybe just create a new function `LvArray::tensorOps:: Rij_eq_AiBj+AjBi`, where `Rij` is in Voigt notation?